src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -56,16 +56,17 @@ public EditCommand parse(String args) throws ParseException {
             editClientDescriptor.setPhone(ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get()));
         }
         if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
-            editClientDescriptor.setEmail(ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get()));
+            editClientDescriptor.setEmail(Optional.of(ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get())));
         }
         if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
-            editClientDescriptor.setAddress(ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get()));
+            editClientDescriptor.setAddress(Optional.of(
+                    ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get())));
         }
 
         if (argMultimap.getValue(PREFIX_BIRTHDAY).isPresent()) {
-            LocalDate dateLocalDate = ParserUtil.parseDate(argMultimap.getValue(PREFIX_BIRTHDAY).get());
+            LocalDate dateLocalDate = ParserUtil.parseDate(argMultimap.getValue(PREFIX_BIRTHDAY).get(), "birthday");
```

Comment:
can potentially use something like `Birthday.class` as a parameter instead of `"birthday"`, but this works too

Reply:
good suggestion, will probably change in v1.4